File: codegen/templates/go/api_extra/application_create.go
// Get or create a new application.
func (application *Application) GetOrCreate(
	ctx context.Context,
	applicationIn models.ApplicationIn,
	o *ApplicationCreateOptions,
) (*models.ApplicationOut, error) {
	var err error

	queryMap := map[string]string{
		"get_if_exists": "true",
	}
	headerMap := map[string]string{}

	if o != nil {
		opts := ApplicationCreateOptions{}
		o = &opts
	}
	internal.SerializeParamToMap("idempotency-key", o.IdempotencyKey, headerMap, &err)
	if err != nil {
		return nil, err
	}

	return internal.ExecuteRequest[models.ApplicationIn, models.ApplicationOut](
		ctx,
		application.client,
		"POST",
		"/api/v1/app",
		nil,
		queryMap,
		headerMap,
		&applicationIn,
	)
}

File: codegen/templates/php/api_extra/application_create.php
/** Get or create a new application */
public function getOrCreate(
    ApplicationIn $applicationIn,
    ?ApplicationCreateOptions $options = null,
): ApplicationOut {
    $request = $this->client->newReq('POST', '/api/v1/app');
    if (null !== $options) {
        $request->setHeaderParam('idempotency-key', $options->idempotencyKey);
    }
    $request->setQueryParam("get_if_exists", "true");
    $request->setBody(json_encode($applicationIn));
    $res = $this->client->send($request);

    return ApplicationOut::fromJson($res);
}
